src/components/SignInButton.tsx

```diff
@@ -113,18 +114,47 @@ export const SignInButton = (props: SignInButtonProps) => {
     await Metrics.captureEvent(event);
   };
 
+  const streamToString = async (error: HttpError): Promise<string> => {
+    const reader = error.body?.getReader();
+    let result = '';
+    const decoder = new TextDecoder('utf-8');
+
+    if (!reader) {
+      return result;
+    }
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const { done, value } = await reader.read();
+      if (done) break;
+      result += decoder.decode(value);
+    }
+    return result;
+  };
```

Comment:
This approach seems complicated, here's what I would suggest.
```suggestion
  const errorStreamToString = async (error: HttpError) => {
    return await new Response(error.body).text();
  };
```